torch/ao/quantization/fx/utils.py

```diff
@@ -642,3 +653,51 @@ def get_skipped_module_name_and_classes(
         skipped_module_classes += get_custom_module_class_keys(prepare_custom_config.float_to_observed_mapping)
 
     return skipped_module_names, skipped_module_classes
+
+def _validate_qconfig_against_dtype_config(
+        qconfig: QConfigAny,
+        dtype_config: DTypeConfig,
+        pattern: Pattern):
+    """
+    Validate whether the QConfig satisfies all of the following constraints from the backend:
+        1. QConfig specified a quantization range that falls within the backend's, if any
+        2. QConfig specified a min scale value that is >= the backend's min scale value, if any
+    """
+    def validate_activation_post_process(
+            activation_post_process_ctr: Union[_PartialWrapper, Type[ObserverBase]],
+            dtype_with_constraints: DTypeWithConstraints,
+            debug_string: str):
+        activation_post_process = activation_post_process_ctr()  # type: ignore[call-arg]
+        qconfig_quant_min = getattr(activation_post_process, "quant_min", None)
+        qconfig_quant_max = getattr(activation_post_process, "quant_max", None)
+        # TODO: for now, just use the existing eps value as scale_min. In the future, we should
```

Comment:
`eps` is usually added to the denominator to prevent division by zero, it's not a replacement for `scale_min`, and `scale_min` is not a replacement for `eps`. Not sure what this TODO is asking.

Reply:
I think the `eps` in observer can be renamed to `scale_min`, it's added in this PR: https://github.com/pytorch/pytorch/commit/cfe1a41b011a7a9b9ffa66cb33f156e19b3e3003, and here is the relevant description of eps:
```
* xnnpack has a requantization scale value restriction, 0x1p-32 <=
    requantization_scale < 256.0 where, `requantization_scale = (input_scale
    * kernel_scale) / (output_scale)`. Using this eps (w/ assumed max value
    of 256) is to prevent requantization_scale to go below xnnpack lower
    threshold.
```

Reply:
but it would make sense to do the renaming in a separate PR though, to reduce confusion

Reply:
`eps` sounds like a good name to me, fwiw.  It's a good practice to have an epsilon in the denominator if the other terms can approach zero.

`scale_min` happens to do the same thing as `eps` today, but the user can specify `scale_min` for some other reason.  We would still need `eps` to guard against division by zero.

Reply:
> It's a good practice to have an epsilon in the denominator if the other terms can approach zero.
I think it's OK to have `eps` hardcoded in the code, but it doesn't mean we need to make `eps` configurable, the reason people added `scale_min` is that they want to change the min value for scale when the scale is very small, the purpose is because of some hardware restrictions of the min scale, which happens to have the same implementation as changing `eps` here. I think `scale_min` and `eps` are different use cases due to different purposes.

Reply:
We're saying the same thing :)

Making eps non-configurable sounds good to me.

Reply:
Ok, sounds good. Let's handle that in a separate PR